src/sparseml/pytorch/image_classification/utils/helpers.py

```diff
@@ -397,8 +397,20 @@ def save_model_training(
         name=f"{save_name}.pth",
         arch_key=arch_key,
     )
+
     info_path = os.path.join(save_dir, f"{save_name}.txt")
+    write_validation_results(info_path, val_res, epoch=epoch)
+
+    if not save_message_shown:
+        print(f"Saving model for epoch {epoch} to {save_dir} for {save_name}")
+
 
+def write_validation_results(info_path, val_res, epoch=0):
```

Comment:
let's make epoch optional and not write it if it's None (no need to write epoch for `eval.txt`)